### PR TITLE
Allow user to configure kafka request timeout

### PR DESCRIPTION
**Pull request by VictorDenisov**
Sometime I would like to control kafka request timeout independently of connection timeout.
This refactoring extract kafka timeout calculation into an option and provides another option for exact timeout value of kafka operations.

### CODE REVIEW

**Comment by achille-roussel** [comment]
Do you mind providing an example of use cases for this feature? It introduces complexity and public APIs that could be hard to get rid of so I’d like to make sure there are important scenarios that can’t be covered without it.

File: time.go [hunk 1]
```diff
@@ -43,13 +43,25 @@ func deadlineToTimeout(deadline time.Time, now time.Time) time.Duration {
 	return deadline.Sub(now)
 }
 
-func adjustDeadlineForRTT(deadline time.Time, now time.Time, rtt time.Duration) time.Time {
+type KafkaTimeoutOption func(time.Time, time.Time, time.Duration) time.Duration
```
[the remark this answers is not in the document]
Comment (VictorDenisov) [reply time.go]: It's not really a full function-option approach. It's just a parameter that happens to be a function. Without it I will have to specify a data structure for the parameter and then turn it into a function that parses this parameter and makes the decision which looked like more code for achieving the same result. Of course I can rewrite it the way you prefer.

[the remark this answers is not in the document]
Comment (achille-roussel) [reply time.go]: I'm not sure I follow, it seems that it could simply be `KafkaTimeout time.Duration` on the `ConnConfig` type. Wouldn't that work?

**Comment by achille-roussel** [comment]
If what we need is disabling kafka timeouts, how about adding this feature? I'm not super thrilled about the chances for misconfiguration between the read deadline and the kafka timeout, right now it's properly managed so there is no risk of messing it up (but it doesn't support disabling the wait).

It seems that there are no real use case for supporting both blocking and non-blocking fetch requests on a single connection, so having it be an immutable configuration option on the connection seems like a good fit, for example:
```go
type ConnConfig struct {
    ...
    EnableNonBlockingRead bool
}
```

Let me know what you think.

**Comment by VictorDenisov** [comment]
I wanted to give the user a more generic and fine tuned way to control timeouts. By default they are handled the old way if the user decides to control the timeouts himself then he is on his own. Though I have to admin this new approach seems convoluted. Let me mull it over for a couple of days. I'll either offer another approach to timeout configuration or we will go with a boolean flag.

**Pull request closed without merging**